Sources/Kommunicate/Classes/KMConversationService.swift

```diff
@@ -76,28 +76,41 @@ public class KMConversationService: KMConservationServiceable,Localizable {
     public func createConversation(
         conversation: KMConversation,
         completion: @escaping (Response)->()) {
+        
+        let dispatchGroup = DispatchGroup()
 
         if let clientId = conversation.clientConversationId, !clientId.isEmpty {
             self.isGroupPresent(clientId: clientId, completion: { present, channel in
                 if present {
                     let groupID = Int(truncating: channel?.key ?? 0)
-                    let response = Response(success: true, clientChannelKey: clientId, error: nil)
-                    guard let currentAssignee = self.assigneeUserIdFor(groupId: groupID),
-                          let newAssignee = conversation.conversationAssignee,
-                          !newAssignee.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty,
-                          newAssignee != currentAssignee else {
-                        completion(response)
-                        return
+                    var response = Response(success: true, clientChannelKey: clientId, error: nil)
+                    
+                    if let currentAssignee = self.assigneeUserIdFor(groupId: groupID), let newAssignee = conversation.conversationAssignee {
+                        if (!(newAssignee.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty) && newAssignee != currentAssignee) {
+                            dispatchGroup.enter()
+                            self.assignConversation(groupId: groupID, to: newAssignee) { result in
+                                switch result {
+                                case .success:
+                                    dispatchGroup.leave()
+                                case .failure(let error):
+                                    response.error = error
+                                    response.success = false
+                                    dispatchGroup.leave()
+                                }
+                            }
+                        }
                     }
-                    self.assignConversation(groupId: groupID, to: newAssignee) { result in
-                        switch result {
-                        case .success:
-                            completion(response)
-                        case .failure(let error):
-                            let errorResponse = Response(success: false, clientChannelKey: clientId, error: error)
-                            completion(errorResponse)
+                    dispatchGroup.enter()
+                    if self.groupMetadata == nil {
+                        dispatchGroup.leave()
+                    } else {
+                        self.updateGroupMetadata(groupId: NSNumber(value: groupID), channelKey: "", metadata: self.groupMetadata) { result in
+                            dispatchGroup.leave()
```

Comment:
Here we need to set the response(`response = result`) just like we do with error above.

Reply:
Done